### ISSUE TITLE
Incorrect results from studentized_range.sf

### ISSUE THREAD

**Issue by yogesh-dhande** [pyodide/pyodide#4890]
## 🐛 Bug

When running studentized_range.sf in a regular jupyter notebook vs in pyodide, I noticed very different values.

### To Reproduce
Run the following code

```python
from scipy.stats import distributions
distributions.studentized_range.sf(4.5, 3, 12)
```

In a regular Jupyter notebook (python 3.9), this code outputs `0.020003353772350296` but in pyodide, it outputs `1.0`
I created an HTML gist for the pyodide reproduction: https://gist.github.com/yogesh-dhande/ef35e5caa2d7ce6103dc12328ee70ead

### Expected behavior

I expect the same numeric output from both the python and pyodide versions.

### Environment

- Pyodide Version: 0.26.1
- Browser version: Chrome Version 126.0.6478.62 (Official Build) (x86_64)
- Any other relevant information:
- SciPy versions: 1.12.0 in python and 1.11.2 in pyodide

**Comment by yogesh-dhande** [comment]
I dug into it a bit and traced it to `integrate.nquad`. Here's a simple code example to reproduce how the `integrate.nquad` results differ in python vs pyodide.

```python
import numpy as np
from scipy import integrate

def func(x, y):
    return x * y**2

def bounds_y(x):
    return [0, 1 - x]

def bounds_x():
    return [0, 1]

integrate.nquad(func, [bounds_y, bounds_x], full_output=True)
```

**Comment by hoodmane** [comment]
Have you tested it against tip of tree? It's quite possible that #4822 fixed it.

**Comment by hoodmane** [comment]
Indeed this is fixed on the main branch:
Stable:
![image](https://github.com/pyodide/pyodide/assets/8739626/00810fae-a417-464c-8814-c6c1e0259db7)

Latest:
![image](https://github.com/pyodide/pyodide/assets/8739626/86a2d9e0-5ceb-4b07-84e2-b44f808c6ff0)

So I'll close it as resolved. The fix will be released in 0.26.2.
